src/Camera/QGCCameraControl.cc

```diff
@@ -1386,26 +1388,34 @@ QGCCameraControl::_updateRanges(Fact* pFact)
         }
     }
     //-- Update limited range set
-    for (Fact* f: rangesSet.keys()) {
-        f->setEnumInfo(rangesSet[f]->optNames, rangesSet[f]->optVariants);
-        if(!updates.contains(f->name())) {
-            _paramIO[f->name()]->optNames = rangesSet[f]->optNames;
-            _paramIO[f->name()]->optVariants = rangesSet[f]->optVariants;
-            emit f->enumsChanged();
-            qCDebug(CameraControlVerboseLog) << "Limited set of options for:" << f->name() << rangesSet[f]->optNames;;
-            updates << f->name();
+    auto setIterator = rangesSet.constBegin();
+    while (setIterator != rangesSet.constEnd()) {
+        const auto &key = setIterator.key();
+        const auto &value = setIterator.value();
+        key->setEnumInfo(value->optNames, value->optVariants);
+        if(!updates.contains(key->name())) {
+            _paramIO[key->name()]->optNames = value->optNames;
+            _paramIO[key->name()]->optVariants = value->optVariants;
+            emit key->enumsChanged();
+            qCDebug(CameraControlVerboseLog) << "Limited set of options for:" << key->name() << value->optNames;
+            updates << key->name();
         }
+        ++setIterator;
     }
     //-- Restore full range set
-    for (Fact* f: rangesReset.keys()) {
-        f->setEnumInfo(_originalOptNames[rangesReset[f]], _originalOptValues[rangesReset[f]]);
-        if(!updates.contains(f->name())) {
-            _paramIO[f->name()]->optNames = _originalOptNames[rangesReset[f]];
-            _paramIO[f->name()]->optVariants = _originalOptValues[rangesReset[f]];
-            emit f->enumsChanged();
-            qCDebug(CameraControlVerboseLog) << "Restore full set of options for:" << f->name() << _originalOptNames[f->name()];
-            updates << f->name();
+    auto resetIterator = rangesReset.constBegin();
+    while (resetIterator != rangesReset.constEnd()) {
+        const auto &key = resetIterator.key();
+        const auto &value = resetIterator.value();
+        key->setEnumInfo(_originalOptNames[value], _originalOptValues[value]);
+        if(!updates.contains(key->name())) {
+            _paramIO[key->name()]->optNames = _originalOptNames[value];
+            _paramIO[key->name()]->optVariants = _originalOptValues[value];
+            emit key->enumsChanged();
+            qCDebug(CameraControlVerboseLog) << "Restore full set of options for:" << key->name() << _originalOptNames[key->name()];
+            updates << key->name();
         }
+        ++resetIterator;
```

Comment:
way way worse to read.  (and the old code was also bad to read).
Please don't just try to make things faster, but they should be faster and better in readability.
Perhaps this code should change to std::map / std::unordered_map as the iteration thru range-based-for has key and value?
to we *really* need key and value?
you see that you are double-searching values based on keys just like the old code?